lib/services/database.go

```diff
@@ -133,6 +133,13 @@ func NewDatabaseFromRDSInstance(instance *rds.DBInstance) (types.Database, error
 
 // NewDatabaseFromRDSCluster creates a database resource from an RDS cluster (Aurora).
 func NewDatabaseFromRDSCluster(cluster *rds.DBCluster) (types.Database, error) {
+	// Aurora Serverless (v1 and v2) does not support IAM authentication
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html#aurora-serverless.limitations
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless-2.limitations.html
+	if aws.StringValue(cluster.EngineMode) == RDSEngineModeServerless {
+		return nil, trace.NotImplemented("unsupported Aurora engine mode %v", RDSEngineModeServerless)
```

Comment:
How/where is this "not implemented" error handled? Does it produce an error in the logs when a serverless instance is encountered?

Reply:
It's currently called by the rds watcher

https://github.com/gravitational/teleport/blob/c9ee98b6af14bed973989e8270c3edf39db8a064/lib/srv/db/cloud/watchers/rds.go#L145

An `info` log is printed when this error happens, I put one sample log in the description

Reply:
Hmm I see. To be honest, I think that instead of returning an error here, it would be more "graceful" to just skip over serverless instances when we're iterating over all fetched Aurora clusters and just not even attempt to convert, since this is a somewhat expected scenario.

Then we can also print a nice message for users e.g. something like:

```go
for _, cluster := range clusters {
  if aws.StringValue(cluster.EngineMode) == RDSEngineModeServerless {
    f.log.Infof("Aurora cluster %q is serverless which doesn't support IAM authentication. Skipping.")
    continue
  }
  ...
}
```

WDYT?

Reply:
Ya. I think that's better. Will update